### PR TITLE
repl config path

### PR DESCRIPTION
**Pull request by alexrudd2**
`mypy` is not happy with the option `modbus_config` being used for both a Path and a dictionary (possibly) loaded from that path.  This fixes another 9 errors.

Since this is technically an interface change, I've kept it separate from my other type fixes.  Another option to to use e.g. `modbus_config_dict` but that seemed verbose.
```python
if modbus_config:
    with open(modbus_config) as my_file:  # pylint: disable=unspecified-encoding
        modbus_config_dict = json.load(my_file)
```
@dhoomakethu

### CODE REVIEW

**Review by janiversen — APPROVED** [review]
LGTM, but please fix “ # pylint: disable=unspecified-encoding” while you are at in (copied from the description of this PR.

I am also very unhappy about the SQL-stubs, its an extra requirement that really are not needed.

**Comment by dhoomakethu** [comment]
LGTM. Thanks

**Review by dhoomakethu — APPROVED** [review]

**Comment by janiversen** [comment]
Added commit to:
- add mypy==1.0.1 (but currently do not activate it in CI or check_ci)
- Remove SQLAlchemyStubs.

If a library do not contain type hints, we should not add extra requirements just to cover that. We use our own version of async-pyserial that too comes with a level of type hints.

The sqlalchemystub slipped through my review, hence correcting it.

**Comment by janiversen** [comment]
Added commit to solve black/pylint.

**Comment by janiversen** [comment]
Added commit with more comments in mixin, to explain the use of execute().

Mypy needs to be silenced for the execute function (not all the others).

@alexrudd2 I will leave it up to you to correct or merge this, since I have corrected it (You have 2 approvals).

**Comment by alexrudd2** [comment]
Screenshot 2023-02-19 at 12 02 06 PM

It seems there is an automatic rule to change PR targets which caused some confusion.  I intended this to be merged into #1356 Sorry

Anyways, I rebased it to squash the lint commits.

**Comment by alexrudd2** [comment]
This is totally unrelated to https://github.com/pymodbus-dev/pymodbus/issues/1357 since I do not yet have a solution to it, so I removed that closing keyword.

**Comment by alexrudd2** [comment]
> @alexrudd2 I will leave it up to you to correct or merge this, since I have corrected it (You have 2 approvals).

I think this PR is now ready.  However, `dev` is a protected branch, and I'm not authorized to merge into it.

**Comment by janiversen** [comment]
I thought you were, I will take a look on that later. The protection should only require an approval, which you have.

**Pull request merged**
